shell/server/grain-server.js

```diff
@@ -273,12 +273,83 @@ Meteor.publish("grainSize", function (grainId) {
   });
 });
 
+const GRAIN_DELETION_MS = 1000 * 60 * 60 * 24 * 30; // thirty days
+SandstormDb.periodicCleanup(86400000, () => {
+  const trashExpiration = new Date(Date.now() - GRAIN_DELETION_MS);
+  globalDb.removeApiTokens({ trashed: { $lt: trashExpiration } });
+  Grains.find({ trashed: { $lt: trashExpiration } }).forEach((grain) => {
+    Grains.remove({ _id: grain._id });
+    globalDb.removeApiTokens({
+      grainId: grain._id,
+      $or: [
+        { owner: { $exists: false } },
+        { owner: { webkey: null } },
+      ],
+    });
+
+    if (grain.lastUsed) {
+      DeleteStats.insert({
+        type: "grain",  // Demo grains can never never get here!
+        lastActive: grain.lastUsed,
+        appId: grain.appId,
+      });
+    }
+
+    waitPromise(globalBackend.deleteGrain(grain._id, grain.userId));
```

Comment:
For robustness, this call should probably happen before removing the table entry. Otherwise, a badly-timed crash could lead to the grain storage still existing with no way to clean it up.

Probably we should make the same change for the explicit delete flow, too...